prf.go

```diff
@@ -86,6 +86,18 @@ func prfPreMasterSecret(publicKey, privateKey []byte, curve namedCurve) ([]byte,
 		resultBytes := result.Bytes()
 		copy(preMasterSecret[len(preMasterSecret)-len(resultBytes):], resultBytes)
 		return preMasterSecret, nil
+	case namedCurveP384:
```

Comment:
Is it possible to combine `namedCurveP256` and `namedCurveP384`, they are pretty much a dupe of each other. Just to make reading through code a little easier/less change of copying bugs.

Reply:
Abstracted better! Same comment as below, LMK if you see any other improvements to be made to the abstraction